src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,38 +205,102 @@ function updateStatsCheckBox() {
+ *         collapse panels button.
+ */
+function expandOrCollapsePanels(element, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    var targetPanels = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(STRING_EXPAND);
+
+    if (isElementAnExpandButton) {
+        expandPanels(element, targetPanels);
+        replaceButtonHtmlAndTooltipText(element, STRING_EXPAND, STRING_COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
-            }
+        collapsePanels(targetPanels);
+        replaceButtonHtmlAndTooltipText(element, STRING_COLLAPSE, STRING_EXPAND);
+    }
+}
+
+/**
+ * Expands all panels. Loads panel data if missing.
+ * @param {DOM} element - The element that was clicked to invoke {@code #expandOrCollapsePanels}
+ */
+function expandPanels(element, panels) {
+    // The expand/collapse button on AJAX-loaded panels is of id collapse-panels-button
+    var areAjaxLoadedPanels = $(element).is($('#collapse-panels-button'));
+    var BASE_TIME_OUT_UNIT_IN_MILLI_SECOND = 50;
+
+    for (var idx = 0; idx < panels.length; idx++) {
+        expandPanel(panels[idx], idx * BASE_TIME_OUT_UNIT_IN_MILLI_SECOND, areAjaxLoadedPanels);
+    }
+}
+
+/**
+ * Expands {@param panel}. If the panel data is not loaded, load it by Ajax.
+ * @param {int} timeOut - is in milliseconds
+ */
+function expandPanel(panel, timeOut, isAjaxLoadedPanel) {
+    var isPanelAlreadyExpanded = $(panel).hasClass('in');
+    if (isPanelAlreadyExpanded) {
+        return;
+    }
+
+    if (isAjaxLoadedPanel) {
+        // Might need to load the panel data by Ajax.
+        var $elementToClickForAjaxLoading = getElementToClickForAjaxLoading(panel);
+        var needToLoadThePanelDataByClickingOnElement = $elementToClickForAjaxLoading.length !== 0;
+        if (needToLoadThePanelDataByClickingOnElement) {
+            // When clicked, the panel data is loaded, and ajax_auto or ajax-response-auto class will
+            // be removed from the element.
+            $elementToClickForAjaxLoading.click();
+
+            // Panel is already expanded as a result of clicking.
+            return;
         }
-        var htmlStr = $(e).html();
-        htmlStr = htmlStr.replace(collapse, expand);
-        $(e).html(htmlStr);
-        var tooltipStr = $(e).attr('data-original-title').replace(collapse, expand);
-        $(e).attr('title', tooltipStr).tooltip('fixTitle').tooltip('show');
     }
+
+    // expand this panel
+    setTimeout(showSingleCollapse, timeOut, panel);
+}
+
+/**
+ * @return {DOM} the element that has class ajax_auto or ajax-response-auto(not both). The element can be
+ *         clicked to load the {@param panel} content by Ajax.
```

Comment:
However, I do not think we should use {@code ajax_auto} here as this ajax_auto does not refer to any code.